src/python/WMCore/Lexicon.py

```diff
@@ -656,3 +667,24 @@ def activity(candidate):
     if candidate in dashboardActivities:
         return True
     raise AssertionError("Invalid dashboard activity: %s should 'test'" % candidate)
+
+
+def getStringsBetween(start, end, source):
+    """
+    get the string between start string and end string for given source string
+    source string is one line string (no new line charactor in it)
+    """
+    regex = r"\%s(.*?)\%s" % (start, end)
+    result = re.match(regex, source)
+    if result:
+        return result.group(1).strip()
+    else:
+        return None
+
+
+def getIterMatchObjectOnRegexp(filePath, regexp):
+    with open(filePath, 'r') as f:
+        mm = mmap.mmap(f.fileno(), 0, access=mmap.ACCESS_READ)
```

Comment:
Just to make sure I understood this mmap, is it really just a mapping (so memory contains position/address of where the file content is) or it actually loads the data into memory?

Reply:
It is the first case(I guess. :-)). It uses virtual memory. I am not sure how much it load to physical memory at one time (I guess it is depending on operating system)
Here is reference I got for this code.

https://stackoverflow.com/questions/14289421/how-to-use-mmap-in-python-when-the-whole-file-is-too-big